src/logging/monitor.ts

```diff
@@ -0,0 +1,84 @@
+import Database from '../db';
+
+// Flush every 10 mins
+const DB_FLUSH_INTERVAL = 10 * 60 * 1000;
+
+/**
+ * This class provides a wrapper around the database for
+ * saving the last timestamp of a container
+ */
+export class LogMonitor {
+	private timestamps: { [containerId: string]: number } = {};
+	private writeRequired: { [containerId: string]: boolean } = {};
+
+	public constructor(private db: Database) {
+		setInterval(() => this.flushDb(), DB_FLUSH_INTERVAL);
+	}
+
+	public updateContainerSentTimestamp(
+		containerId: string,
+		timestamp: number,
+	): void {
+		this.timestamps[containerId] = timestamp;
+		this.writeRequired[containerId] = true;
+	}
+
+	public async getContainerSentTimestamp(containerId: string): Promise<number> {
```

Comment:
This function isn't safe to be called multiple times in parallel with the same `containerId` when this.timestamps[containerId] is null btw

Reply:
Whilst this should be completely fine in the way that it's used, I've added an `this.timestamps[containerId] = 0` before ever `await`ing on the database calls. This *could* make some logs be sent more than once, but we're talking the time it takes to write to the database, so one or two absolute max (assuming nothing else is going on).